seqr/views/apis/igv_api.py

```diff
@@ -126,9 +132,43 @@ def fetch_igv_track(request, project_guid, igv_track_path):
     if igv_track_path.endswith('.bam.bai') and not does_file_exist(igv_track_path, user=request.user):
         igv_track_path = igv_track_path.replace('.bam.bai', '.bai')
 
+    if is_google_bucket_file_path(igv_track_path):
+        return _stream_gs(request, igv_track_path)
+
     return _stream_file(request, igv_track_path)
 
 
+def _stream_gs(request, gs_path):
+    headers = _get_gs_rest_api_headers(request.META.get('HTTP_RANGE'), gs_path, user=request.user)
+
+    response = requests.get(
+        'https://storage.googleapis.com/{}'.format(gs_path.replace('gs://', '', 1)),
+        headers=headers,
+        stream=True)
+
+    return StreamingHttpResponse(response.iter_content(chunk_size=65536), status=response.status_code,
+                                 content_type='application/octet-stream')
+
+
+def _get_gs_rest_api_headers(range_header, gs_path, user=None):
+    headers = {'Authorization': 'Bearer {}'.format(_get_access_token(user))}
+    if range_header:
+        headers['Range'] = range_header
+    headers['x-goog-user-project'] = get_google_project(gs_path)
```

Comment:
If there is no project we should not send the header, sending None in the header is bad practice

Reply:
I thought the tool (package) would remove None headers.